deltachat-ffi/src/providers.rs

```diff
@@ -0,0 +1,93 @@
+use deltachat::dc_tools::{as_str, dc_strdup, StrExt};
+extern crate deltachat_provider_overview;
+use deltachat_provider_overview::StatusState;
+use std::ptr;
+
+#[no_mangle]
+pub type dc_provider = deltachat_provider_overview::Provider;
+
+#[no_mangle]
+pub unsafe extern "C" fn dc_get_json_provider_info_from_email(
+    email: *const libc::c_char,
+) -> *mut libc::c_char {
+    let option = deltachat_provider_overview::get_provider_info(
+        deltachat_provider_overview::get_domain_from_email(as_str(email)),
+    );
+    if let Some(provider) = option {
+        serde_json::to_string(provider.0).unwrap().strdup()
+    } else {
+        dc_strdup(ptr::null())
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn dc_get_provider_info_from_email(
+    email: *const libc::c_char,
+) -> *const dc_provider {
+    let option = deltachat_provider_overview::get_provider_info(
+        deltachat_provider_overview::get_domain_from_email(as_str(email)),
+    );
+    if let Some(provider) = option {
+        provider.0
```

Comment:
Why does this have to be `.0`?

Reply:
get_provider_info returns `Option<(&Provider, Vec<&'static str>)>` (thinking about it, I do not know why; the second part is a list of domains that belong to this provider but why should we need this?), we only want the &Provider part. `.0` returns the first part of a tuple.

Reply:
I guess we could expose a separate function for this.

Reply:
Is there any reason why we need the domain part at all?

Reply:
For information it might be useful to show it, I would say we move the domain part to a separate function in the provider overview, we don't need to include it in the deltachat-ffi yet, but if we need it in the future its already there.